src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java

```diff
@@ -262,11 +290,20 @@ public void validate() throws AppEngineException {
           "Validation Error: Java Tools jar location '"
               + JAR_LOCATIONS.get(JAVA_TOOLS_JAR) + "' is not a file.");
     }
+    try {
+      if (!isComponentInstalled("app-engine-java")) {
```

Comment:
:+1: to @meltsufin's comment: I tried running `gcloud` from a VM shared drive and it was glacially sloooow.  If JAVA_TOOLS_JAR (or one of the jars) isn't available then we're either dealing with a missing `app-engine-java` component, or `appengine-plugins-core` is out of sync with the installed version and